### PR TITLE
Adds world.address/port/url implementation

### PR DESCRIPTION
**Pull request by ZeWaka**
No setting yet - plan to do setting of port/address via DreamServer arguments in separate PR

### CODE REVIEW

**Review by wixoaGit — COMMENTED** [review]
`world.address` seems to always be 0.0.0.0, while in byond it's my computer's local IP address.

**Comment by ZeWaka** [comment]
> the reason it's showing up as`0.0.0.0` is because `IPAddress.Any` which the default set, goes to `0.0.0.0` instead of BYOND, which finds a different one - ex. for me `172.20.48.0`

I think this is correct. It's supposed to get your local IP, and the world is being hosted on `0.0.0.0` by default currently.

Also, `world.internet_address` will be a pain, since we'll have to open a socket to an external server (say 8.8.8.8 or something).

File: OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs [hunk 1]
```diff
@@ -59,7 +59,11 @@ class DreamMetaObjectWorld : DreamMetaObjectRoot {
             } else if (variableName == "maxz") {
                 return new DreamValue(Program.DreamMap.Levels.Count);
             } else if (variableName == "address") {
-                return DreamValue.Null;
+                return new(Program.DreamServer.Address.ToString());
+            } else if (variableName == "port") {
+                return new(Program.DreamServer.Port.ToString());
+            } else if (variableName == "url") {
+                return new("byond://" + Program.DreamServer.Address + ":" + Program.DreamServer.Port);
```
Comment (wixoaGit) [inline OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs:66]: I'm hesitant to have this return a `byond://` url

File: OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs [hunk 2]
```diff
@@ -59,7 +59,11 @@ class DreamMetaObjectWorld : DreamMetaObjectRoot {
             } else if (variableName == "maxz") {
                 return new DreamValue(Program.DreamMap.Levels.Count);
             } else if (variableName == "address") {
-                return DreamValue.Null;
+                return new(Program.DreamServer.Address.ToString());
+            } else if (variableName == "port") {
+                return new(Program.DreamServer.Port.ToString());
```
Comment (wixoaGit) [inline OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs:64]: `/world.port` is a number

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs (hunk 1 above)
Comment (ZeWaka) [reply OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs:66]: fair, was just replicating existing behavior - we should probably specify our own thing

File: OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs (hunk 2 above)
Comment (ZeWaka) [reply OpenDreamServer/Dream/Objects/MetaObjects/DreamMetaObjectWorld.cs:64]: 😹

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**